### ISSUE TITLE
ClickHouse ERROR: Distance function argument of Annoy index must be of type String.

### ISSUE THREAD

**Issue by sheiy** [hwchase17/langchain#6208]
### System Info

langchain: 0.0.200
platform: macOS
python:  3.10.11
clickhouse: version 23.5.2.7 (official build)

### Who can help?

@hwchase17 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

<img width="1351" alt="image" src="https://github.com/hwchase17/langchain/assets/12210038/187c268c-7eca-40f1-80a6-596ba3885be9">

### Expected behavior

success write data to clickhouse

**Comment by devstein** [comment]
_Generated by a 🤖_

### Anwser

To resolve the issue, update the `index_param` in `ClickHouseSettings` to use the `metric` parameter as a string:
```py
index_param: Optional[Union[List, Dict]] = [100, f"'{metric}'"]
```
Also, ensure the `metric` parameter in the `Annoy.from_texts()` function is set to a string value, such as "angular" or "euclidean":

```python
vector_store_v2 = Annoy.from_texts(
    texts, embeddings_func, metric="angular", n_trees=100, n_jobs=1
)
```

### Sources
- [langchain/vectorstores/clickhouse.py](https://github.com/hwchase17/langchain/blob/master/langchain/vectorstores/clickhouse.py)
- [docs/modules/indexes/vectorstores/examples/annoy.ipynb](https://github.com/hwchase17/langchain/blob/master/docs/modules/indexes/vectorstores/examples/annoy.ipynb)

---

This response is meant to be helpful and save users time. It is not meant to be a precise solution.

Please leave a 👍 if this is helpful and 👎 if it is irrelevant

**Comment by sheiy** [comment]
@haoch

**Comment by haoch** [comment]
@sheiy Will test with the clickhouse version you used and get back to you shortly.

**Comment by haoch** [comment]
@sheiy It will be fixed with #6385
